Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EndOfYearDataManager.swift

```diff
@@ -318,6 +322,32 @@ class EndOfYearDataManager {
 
         return numberOfItemsInListeningHistory
     }
+
+    /// Returns the approximate listening time for the current year
+    func yearOverYearListeningTime(dbQueue: FMDatabaseQueue) -> YearOverYearListeningTime {
+        var listeningTimeThisYear: Double = 0
+        var listeningTimePreviousYear: Double = 0
+
+        dbQueue.inDatabase { db in
+            do {
+                let query = "SELECT DISTINCT \(DataManager.episodeTableName).uuid, SUM(playedUpTo) as totalPlayedTime from \(DataManager.episodeTableName) WHERE \(listenedEpisodesThisYear) UNION SELECT DISTINCT \(DataManager.episodeTableName).uuid, SUM(playedUpTo) as totalPlayedTime from \(DataManager.episodeTableName) WHERE \(listenedEpisodesPreviousYear)"
```

Comment:
If the user doesn't have any listening history for the previous year, then the result will be:

Note: I modified the query to return the label to show the order better.

![Screenshot 2023-10-31 at 2 46 14 PM](https://github.com/Automattic/pocket-casts-ios/assets/793774/de5bf10b-d4a1-46a8-a85a-af3fdc95b32e)

When using `UNION` the order of the queries is not preserved, so it's not guaranteed the first item will always be 2023, and the second 2022. 

Using to using `UNION ALL` will preserve the order regardless of the values:


![Screenshot 2023-10-31 at 2 56 58 PM](https://github.com/Automattic/pocket-casts-ios/assets/793774/8c45052a-854f-4e50-984a-12852d512e55)


---

```suggestion
                let query = "SELECT DISTINCT \(DataManager.episodeTableName).uuid, SUM(playedUpTo) as totalPlayedTime from \(DataManager.episodeTableName) WHERE \(listenedEpisodesThisYear) UNION ALL SELECT DISTINCT \(DataManager.episodeTableName).uuid, SUM(playedUpTo) as totalPlayedTime from \(DataManager.episodeTableName) WHERE \(listenedEpisodesPreviousYear)"
```